whole-tale/wt_versioning
# Restore Tale to a given version if id is provided

If `versionId` was provided in `POST /tale/:id/copy` restore the copy to corresponding version.

### How to test?
1. Depends on:
    * https://github.com/whole-tale/girder_wholetale/pull/536
    * https://github.com/whole-tale/wt_versioning/pull/42 (branched of it, so checking `refactor` branch is enough to satisfy this)
2. Create a Tale using JupyterLab env, upload file `filename1.txt` to workspace
3. Save a version of the Tale
4. Change env to RStudio, rename `filename1.txt` to `something.else`.
5. Using Swagger: `POST /tale/:id/copy` without providing `versionId`.
6. You should get a copied Tale with env Rstudio and `something.else` in the workspace.
7. Using Swagger `POST /tale/:id/copy` with `versionId` from step 3.
8. You should get a copied Tale with env JupyterLab and `filename1.txt` in the workspace.
9. OPTIONAL: create more than one version and pick one "in the middle".
